doc/man7/provider-store.pod

```diff
@@ -0,0 +1,225 @@
+L<OSSL_STORE_INFO(3)/SUPPORTED OBJECTS>.
+
+=item "subject" (B<OSSL_STORE_PARAM_SUBJECT>) <octet string>
+
+Indicates that the caller wants to search for an object with the given
+subject associated.  This can be used to select specific certificates
+by subject.
+
+The contents of the octet string is expected to be in DER form.
+
+=item "issuer" (B<OSSL_STORE_PARAM_ISSUER>) <octet string>
+
+Indicates that the caller wants to search for an object with the given
+issuer associated.  This can be used to select specific certificates
+by issuer.
+
+The contents of the octet string is expected to be in DER form.
+
+=item "serial" (B<OSSL_STORE_PARAM_SERIAL>) <integer>
+
+Indicates that the caller wants to search for an object with the given
+serial number associated.
+
+=item "digest" (B<OSSL_STORE_PARAM_DIGEST>) <utf8 string>
+
+=item "fingerprint" (B<OSSL_STORE_PARAM_FINGERPRINT>) <octet string>
+
+Indicates that the caller wants to search for an object with the given
+fingerprint, computed with the given digest.
+
+=item "alias" (B<OSSL_STORE_PARAM_ALIAS>) <utf8 string>
+
+Indicates that the caller wants to search for an object with the given
+alias (some call it a "friendly name").
+
+=back
+
+Several of these search criteria may be combined.  For example, to
+search for a certificate by issuer+serial, both the "issuer" and the
+"serial" parameters will be given.
+
+=head2 Object Parameters
+
+The callback that's called by OP_store_load() should receive some of
+these parameters:
+
+=over 4
+
+=item "type" (B<OSSL_STORE_PARAM_TYPE>) <integer>
+
+The number is one of the object types described in
+L<OSSL_STORE_INFO(3)/SUPPORTED OBJECTS>.
+
+=item "name" (B<OSSL_STORE_PARAM_NAME>) <utf8 string>
+
+=item "desc" (B<OSSL_STORE_PARAM_DESC>) <utf8 string>
+
+If the type given by the "type" parameter is B<OSSL_STORE_INFO_NAME>,
+"name" and "desc" contain the full URI of an object container and the
+extra description.  For the B<OSSL_STORE_INFO_NAME> type, the "name"
+parameter must be given, while the "desc" parameter is optional.
+
+=item "algorithm" (B<OSSL_STORE_PARAM_ALGORITHM>) <utf8 string>
+
+if the type given by the "type" parameter is B<OSSL_STORE_INFO_PARAMS>
+or B<OSSL_STORE_INFO_PKEY>, this parameter must also be given, to
+inform the OpenSSL library what sort of key it's being passed.
+
+=item "reference" (B<OSSL_STORE_PARAM_REFERENCE>) <octet string>
+
+For a STORE implementation that exists in the same provider as a
+L<openssl-keymgmt(7)> implementation and works in tandem with it, this
+parameter can be used as an identity to be passed to
+OP_keymgmt_loaddomparams() or OP_keymgmt_loadkey().
+
+=item "rawdata" (B<OSSL_STORE_PARAM_RAWDATA>) <octet string>
+
+=item "rawencoding" (B<OSSL_STORE_PARAM_RAWENC>) <utf8 string>
+
+A STORE implementation may pass an object in raw encoded form in the
+"rawdata" form, and indicate the encoding with "rawencoding".
+Currently, the only supported encoding is "DER".
```

Comment:
Do we specifically reject non-DER BER/etc.?

Reply:
There are certain non-DER features that we don't support, yeah

Reply:
That's not a terribly confidence-inspiring answer -- if we claim to only support "DER" I would interpret that to be "strict DER" and that we reject *all* non-DER features, not just some of them.

Reply:
Well, strictly speaking, we will support whatever the d2i functions support.  I *think* it's strict DER, but am not 100% sure.